vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/grid/it/GridSetItemsAfterDetachPage.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2000-2020 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.grid.it;
+
+import java.util.Arrays;
+import java.util.List;
+
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.grid.Grid;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.router.Route;
+
+@Route
+public class GridSetItemsAfterDetachPage extends VerticalLayout {
+
+
+    public GridSetItemsAfterDetachPage() {
+        final List<String> items = Arrays.asList("basdlkf", "laksdfh");
+        final Grid<String> grid = new Grid<>();
+        final Button back = new Button("back");
+        back.setId("back");
+        grid.addColumn(e -> e).setHeader("bla");
+        grid.addSelectionListener(e -> {
+            remove(grid);
+            add(back);
+        });
+        grid.setItems(items);
+        add(grid);
+        back.addClickListener(e -> {
+            this.remove(back);
+            grid.setItems(items);
+            this.add(grid);
+        });
+    }
+
+}
```

Comment:
Code style nits:
- new line to end of file
- there's double empty lines below `class`